d2layouts/d2sequence/sequence_diagram.go

```diff
@@ -20,33 +20,71 @@ type sequenceDiagram struct {
 	lifelines []*d2graph.Edge
 	actors    []*d2graph.Object
 	spans     []*d2graph.Object
+	notes     []*d2graph.Object
 
 	// can be either actors or spans
 	// rank: left to right position of actors/spans (spans have the same rank as their parents)
 	objectRank map[*d2graph.Object]int
 
 	// keep track of the first and last message of a given actor/span
 	// the message rank is the order in which it appears from top to bottom
+	// TODO rank is used for y coordinate here and x coordinate above??
```

Comment:
this is gone after the self edges PR.
Anyway, rank is just another word for `order`, so actors rank is left to right, while edges rank is top to bottom